main.go

```diff
@@ -61,6 +61,8 @@ func main() {
 	var metricsAddr string
 	var enableLeaderElection bool
 	var probeAddr string
+	var watchNamespace string
+	flag.StringVar(&watchNamespace, "watch-namespace", os.Getenv("WATCH_NAMESPACE"), "The namespace to watch, if not specified, all namespaces will be watched")
```

Comment:
If `WATCH_NAMESPACE` is not specified, will `&watchNamespace` point to an invalid address? It may be better to provide `""` as the default value.

Reply:
Actually, the zero value of string is "", see https://go.dev/play/p/VxuM1JP4nxI.
watchNamespace  := ""  is the same as var watchNamespace string